### DISCUSSION TITLE
Help with v2 designer

### DISCUSSION THREAD

**Discussion by tznind** [tui-cs/Terminal.Gui#3474] [Q&A]
If anyone has the time to spare I would really appreciate some help getting TerminalGuiDesigner to work correctly with the latest v2 changes.

It was stable(ish) at `2.0.0-pre.250` (which is where the `v2` branch is at) but I have been unable to get through all the work from breaking changes to keep it up to date since.  

I have a building solution in 
https://github.com/gui-cs/TerminalGuiDesigner/pull/289

But 243/1135 failing tests (mostly around key handling and mouse positions).

Any help would be appreciated, even just reviewing that PR and pointing out any of the changes that are not the correct 'drop in' replacement for removed APIs.

**Comment by tig** [comment]
I just dove in.

I'm going to recommend we get #3482 merged asap as it will enable you to remove a bunch of sketchy code from `PosExtensions` and `PosTypes` that is no longer needed.

**Reply by dodexahedron** [reply]
Plus there are other API surface changes and enhancements in the pipeline that will make his life easier as well that I'd hate for him to spend extra effort before they're here.

In particular, back-defining some interfaces should be a boon for TGD to simplify stuff, even if and when reflection is still used, but should also reduce the need for reflection further and make generics way easier, anyway.
